tests/Google/Ads/GoogleAds/Lib/V1/StatusMetadataExtractorTest.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Ads\GoogleAds\Lib\V1;
+
+use Google\Ads\GoogleAds\V1\Errors\AdGroupErrorEnum\AdGroupError;
+use Google\Ads\GoogleAds\V1\Errors\CampaignErrorEnum\CampaignError;
+use Google\Ads\GoogleAds\V1\Errors\ErrorCode;
+use Google\Ads\GoogleAds\V1\Errors\GoogleAdsError;
+use Google\Ads\GoogleAds\V1\Errors\GoogleAdsFailure;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * Unit tests for `StatusMetadataExtractor`.
+ *
+ * @covers \Google\Ads\GoogleAds\Lib\V1\StatusMetadataExtractor
+ * @small
+ */
+class StatusMetadataExtractorTest extends TestCase
+{
+    /** @var StatusMetadataExtractor $statusMetadataExtractor */
+    private $statusMetadataExtractor;
+
+    /**
+     * @see \PHPUnit\Framework\TestCase::setUp()
+     */
+    protected function setUp()
+    {
+        $this->statusMetadataExtractor = new StatusMetadataExtractor();
+    }
+
+    public function testExtractGoogleAdsFailure()
+    {
+        $expected = new GoogleAdsFailure();
```

Comment:
The creation of this object is done at least two times in the tests (`testExtractGoogleAdsFailure ` and `testExtractErrorMessageList `), could you factorize?

Ditto in V2 and V3.